### PR TITLE
Admin spawnable version of the Death Ripley that actually inflicts damage and can rip arms off

### PR DESCRIPTION
**Pull request by Irafas**
:cl: 
add: Admin spawnable Death Ripley that comes with a version of the Death Clamp that actually inflicts damage and rips arms off.
/:cl:

The Death Ripley uses a KILL CLAMP rather than the normal clamp and overrides its action, though doesn't actually inflict any damage on the target. This fix simply inflicts that damage.

~~On disarm intent the KILL CLAMP is supposed to rip off the targets arms. As of writing this I'm not sure how to do this. I've been looking through trying to find how to do this, but any hints on where to look would be appreciated.~~
figured it out.

### CODE REVIEW

**Comment by Irafas** [comment]
For some reason if only the left arm is missing then the right arm won't be ripped off, but if the right arm is missing then the left one is. It has something to do with the order of them in the code, but i'm not sure why. I also need to have it add to the log when it rips off a persons arms.

**Comment by SpaceManiac** [comment]
If the left arm is missing, `get_bodypart` will return null and the `affected.dismember` call will crash the proc.

**Comment by Irafas** [comment]
Alright so this bug fix should be fixed now. The Kill Clamp now inflicts damage on the target and if the user is in disarm intent it now rips off the victims arms.

_**RIP'N'TEAR**_

**Comment by RandomMarine** [comment]
Wasn't that always an intended joke?

**Comment by Jordie0608** [comment]
Yes the lack of damage is intentional.

**Comment by Irafas** [comment]
Well I guess this was a waste of time. Still it made for some practice. If people want to keep the old joke clamps I'll close this PR.

**Comment by Jordie0608** [comment]
You could change this into another subtype of clamp that does actual damage if you wanted.

**Comment by Irafas** [comment]
Alright so now there is a "proper" version of the death clamp that does actually inflict damage, and a "proper" version of the death ripley that comes equipped with it. The normal death ripley performs exactly like it did before and the new one is currently only spawnable by admins.

File: code/game/mecha/equipment/tools/work_tools.dm [hunk 1]
```diff
@@ -108,11 +117,38 @@
...
+				var/obj/item/bodypart/affected = C.get_bodypart(BODY_ZONE_L_ARM)
+				if(affected != null)
+					affected.dismember(damtype)
+					play_sound = TRUE
+				affected = C.get_bodypart(BODY_ZONE_R_ARM)
+				if(affected != null)
+					affected.dismember(damtype)
+					play_sound = TRUE
+				if(play_sound)
+					playsound(get_turf(src), get_dismember_sound(), 80, 1)
```
Comment (ShizCalev) [inline code/game/mecha/equipment/tools/work_tools.dm]: TRUE at the end.

File: code/game/mecha/equipment/tools/work_tools.dm [hunk 2]
```diff
@@ -108,11 +117,38 @@
...
+					target.visible_message("<span class='danger'>[chassis] rips [target]'s arms off.</span>", \
+								   "<span class='userdanger'>[chassis] rips [target]'s arms off.</span>")
+					add_logs(chassis.occupant, M, "attacked", "[name]", "(INTENT: [uppertext(chassis.occupant.a_intent)]) (DAMTYE: [uppertext(damtype)])")
```
Comment (ShizCalev) [inline code/game/mecha/equipment/tools/work_tools.dm]: be explicit about the action that happened instead of just "attacked"

**Review by ShizCalev — DISMISSED** [review]

File: code/game/mecha/equipment/tools/work_tools.dm [hunk 3]
```diff
@@ -78,6 +78,14 @@
 	name = "\improper KILL CLAMP"
 	desc = "They won't know what clamped them!"
 	energy_drain = 0
+	var/real_clamp = FALSE
+	dam_force = 0
```
Comment (Cyberboss) [inline code/game/mecha/equipment/tools/work_tools.dm]: swap the lines of these two

File: code/game/mecha/equipment/tools/work_tools.dm [hunk 4]
```diff
@@ -108,11 +116,41 @@
...
+					affected.dismember(damtype)
+					play_sound = TRUE
+					limbs_gone = ", [affected]"
+				affected = C.get_bodypart(BODY_ZONE_R_ARM)
+				if(affected != null)
+					affected.dismember(damtype)
+					play_sound = TRUE
+					limbs_gone = "[limbs_gone], [affected]"
+				if(play_sound)
+					playsound(get_turf(src), get_dismember_sound(), 80, TRUE)
```
Comment (Cyberboss) [inline code/game/mecha/equipment/tools/work_tools.dm]: playsound(src, ...

File: code/game/mecha/working/ripley.dm [hunk 5]
```diff
@@ -88,6 +88,17 @@
...
 
+/obj/mecha/working/ripley/deathripley/real
+	desc = "OH SHIT IT'S THE DEATHSQUAD WE'RE ALL GONNA DIE. FOR REAL"
+
+/obj/mecha/working/ripley/deathripley/real/Initialize()
+	. = ..()
+	for(var/obj/item/mecha_parts/mecha_equipment/E in equipment)
+		E.detach()
+		qdel(E)
+	var/obj/item/mecha_parts/mecha_equipment/ME = new /obj/item/mecha_parts/mecha_equipment/hydraulic_clamp/kill/real
```
Comment (Cyberboss) [inline code/game/mecha/working/ripley.dm]: equipment.Cut() above this line.

**Review by Cyberboss — DISMISSED** [review]

**Comment by AlexTheSysop42** [comment]
Was the original not only adminspawn?

**Pull request merged**